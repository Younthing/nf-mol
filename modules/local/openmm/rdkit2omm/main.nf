process RDKIT2OMM {
    tag "${meta.id}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"

    input:
    tuple val(meta), path(sdf_file, stageAs: 'input.sdf')

    output:
    tuple val(meta), path("*_omm.pdb"), emit: pdb
    path "versions.yml", emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    def name = task.ext.name ?: "LIG"

    """
    rdkit2omm.py \\
        ${sdf_file} \\
        --name ${name} \\
        --output ${prefix}_omm.pdb \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rdkit: \$(python -c "import rdkit; print(rdkit.__version__)")
        openff: \$(python -c "import openff.toolkit; print(openff.toolkit.__version__)")
        openmm: \$(python -c "import openmm; print(openmm.__version__)")
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}_omm.pdb

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        rdkit: "2024.03.5"
        openff: "0.16.7"
        openmm: "8.2"
    END_VERSIONS
    """
}
