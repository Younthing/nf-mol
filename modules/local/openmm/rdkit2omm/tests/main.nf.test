nextflow_process {
    name "Test Process RDKIT2OMM"
    script "../main.nf"
    process "RDKIT2OMM"

    tag "modules"
    tag "modules_nfcore"
    tag "openmm"
    tag "openmm/rdkit2omm"

    test("Convert SDF to OpenMM PDB") {
        when {
            process {
                """
                input[0] = [
                    [ id:'6w70', single_end:false ],
                    file("${projectDir}/data/openmm/6w70_prepared.sdf")
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path("${process.out.pdb[0][1]}").exists() },
                { assert path("${process.out.versions[0]}").exists() },
                { assert path("${process.out.pdb[0][1]}").getText().contains("HETATM") },
                { assert path("${process.out.versions[0]}").getText().contains("rdkit") },
                { assert path("${process.out.versions[0]}").getText().contains("openff") },
                { assert path("${process.out.versions[0]}").getText().contains("openmm") }
            )
        }
    }

    test("Convert SDF to OpenMM PDB - stub") {
        options "-stub"

        when {
            process {
                """
                input[0] = [
                    [ id:'6w70', single_end:false ],
                    file("${projectDir}/data/openmm/6w70_prepared.sdf")
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() },
                { assert path("${process.out.pdb[0][1]}").exists() },
                { assert path("${process.out.versions[0]}").exists() }
            )
        }
    }
}
