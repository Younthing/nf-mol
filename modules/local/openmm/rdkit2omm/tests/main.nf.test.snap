{
    "Convert SDF to OpenMM PDB - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "6w70",
                            "single_end": false
                        },
                        "6w70_omm.pdb:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,d30eb724813f2bca4e7fd30f986215cb"
                ],
                "pdb": [
                    [
                        {
                            "id": "6w70",
                            "single_end": false
                        },
                        "6w70_omm.pdb:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,d30eb724813f2bca4e7fd30f986215cb"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.4"
        },
        "timestamp": "2025-02-26T13:12:17.038691556"
    }
}